src/Appium.Net/Appium/Android/KeyEvent.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenQA.Selenium.Appium.Android
+{
+    public class KeyEvent
+    {
+        private int? _keyCode; 
+        private int? _metaState;
+        private int? _flags;
+
+        public KeyEvent WithKeyCode(int? androidKeyCode)
+        {
+            _keyCode = androidKeyCode;
+            return this;
+        }
+
+        public KeyEvent WithMetaState(int? metaState)
+        {
+            _metaState = metaState;
+            return this;
+        }
+
+        public KeyEvent WithFlag(int? flag)
+        {
+            _flags = flag;
```

Comment:
if a single flag is added then it must combined with existing flags (e.g. `_flags |= flag`)

Reply:
so it should be:
`_flags |= flag`
? 
Is this the same with _metastate?

Reply:
it depends on how you'd like to design the builder.

Either you decide to add each flag separately and them merge them internally or you decide a client to merge them and then pass to the builder, or both. Same with metastate. What I currently observe is the argument name is confusing to what the method actually does.

Reply:
Hope that makes more sense now